### DISCUSSION TITLE
HyperlinkedModelSerializer doesn't respect SECURE_PROXY_SSL_HEADER settings

### DISCUSSION THREAD

**Discussion by ykursav** [encode/django-rest-framework#9377] [General]
Hello,

I am using django behind docker proxy and after that with nginx reverse proxy (serves as http). Docker proxy wrapping around nginx for https. The issue when I set SECURE_PROXY_SSL_HEADER like below, I am retrieving HyperlinkedModelSerializer url field as http instead of https. In my opinion, expected behaviour should be returning https and respecting that django setting.

`SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")`

## Checklist

- [x] Raised initially as discussion #...
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.

**Comment by lovelydinosaur** [comment]
Thanks I've pitched this over to a discussion, until we've figured if there's an issue here.

Can you start with some debugging here?...

`HyperlinkedModelSerializer` pretty much uses...

```python
path = django.urls.reverse(...)
url = request.build_absolute_uri(path)
```

You probably want to start your debugging with a view that does the above and figure out if there's some mis-configuration taking place.
